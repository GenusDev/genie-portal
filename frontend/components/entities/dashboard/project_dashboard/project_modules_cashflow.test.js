import React from 'react'
import {render, unmountComponentAtNode} from 'react-dom'
import CashFlowGraph from 'Entities/dashboard/project_dashboard/project_modules_cashflow'
import { shallow, mount, render as eRender } from 'enzyme'


// import {setUp} from './project_graph_d3logic';

function componentSetup(){
  const props = {
    fetchProjects: jest.fn(()=>{
      return Promise.resolve();
    }),
    actual_cashflow: {1:10, 2:-30, 3:40, 4:50},
    accum_actual_cashflow: {1:10, 2:-20, 3:20, 4:70},
    projected_cashflow: {1:10, 2:-15, 3:60, 4:80},
    accum_projected_cashflow: {1:10, 2:-5, 3:55, 4:140},
    height: 200,
    width: 300,
  }
  const eWrapper = mount(<CashFlowGraph {...props}/>)
  return {
    props,
    eWrapper
  }
}


describe('Project Module Component', () => {
 // it('the data being rendered should match the number of projects in the database', () => {
 //   const { eWrapper} = setup()
 //   // expect(eWrapper.find('section').hasClass('graph-container')).toBe(true)
 //    expect(eWrapper).toMatchSnapshot();
 // })


   it('should match snapshot', () => {
     const { eWrapper} = componentSetup()
     // expect(eWrapper.find('section').hasClass('graph-container')).toBe(true)
      expect(eWrapper).toMatchSnapshot();
   })




   it('scales should default to the min/max of projected or actual cashflows', () => {
     const {props, eWrapper} = componentSetup()
     const [xAxisScale, yAxisScale, yLinesScale] = eWrapper.instance().defineScales()

     expect(xAxisScale(2)).toEqual(props.width/Object.keys(props.actual_cashflow).length*2)
     expect(yAxisScale(20)).toEqual(-342)
     expect(yLinesScale(20)).toEqual(-24.666666666666686)

   })

})

test('1 + 1 equals 2', () => {
  expect(1 + 3).toBe(4);
});

//
// accum_actual_cashflow = {
//   1: 36974
//   2: 76992
//   3: 93849
//   4: 90934
//   5: 111259
//   6: 119123
//   7: 144483
//   8: 172590
//   9: 0
//   10: 0
//   11: 0
//   12: 0
//   13: 0
//   14: 0
//   15: 0
//   16: 0
//   17: 0
//   18: 0
//   19: 0
//   20: 0
//   21: 0
//   22: 0
//   23: 0
//   24: 0
//   25: 0
//   26: 0
//   27: 0
//   28: 0
// }
