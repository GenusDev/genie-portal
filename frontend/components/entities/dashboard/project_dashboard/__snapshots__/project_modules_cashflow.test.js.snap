// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Module Component should match snapshot 1`] = `
<CashFlowGraph
  accum_actual_cashflow={
    Object {
      "1": 10,
      "2": -20,
      "3": 20,
      "4": 70,
    }
  }
  accum_projected_cashflow={
    Object {
      "1": 10,
      "2": -5,
      "3": 55,
      "4": 140,
    }
  }
  actual_cashflow={
    Object {
      "1": 10,
      "2": -30,
      "3": 40,
      "4": 50,
    }
  }
  fetchProjects={[MockFunction]}
  height={200}
  projected_cashflow={
    Object {
      "1": 10,
      "2": -15,
      "3": 60,
      "4": 80,
    }
  }
  width={300}
>
  <div
    className="cashflow-graph"
  >
    <div
      id="cash-graph"
    >
      <div
        className="title-wrapper"
      >
        <h3
          className="text-hidden"
        >
          cashflow
        </h3>
      </div>
    </div>
  </div>
</CashFlowGraph>
`;
